opensplice_cmake_module/env_hook/opensplice.sh.in

```diff
@@ -1,10 +1,44 @@
-export OSPL_HOME=@OpenSplice_HOME@
+_BUILDTIME_OSPL_HOME=@OpenSplice_HOME@
+
+# Give priority to the OpenSplice installation found when the workspace was built, if it's valid.
+# If it's not valid, the current value of the OSPL_HOME environment variable will be used.
+# The one detected at buildtime may refer to an invalid path for users using "fat archive" binaries, for example.
+if [ -d "$_BUILDTIME_OSPL_HOME" ]; then
+  # Warn about unused environment variable.
+  if [ ! -z "$OSPL_HOME" -a "$OSPL_HOME" != "$_BUILDTIME_OSPL_HOME" ]; then
+    echo "[opensplice_cmake_module] Warning: OSPL_HOME environment variable is set to [[$OSPL_HOME]]." \
+      "When the workspace was built, OpenSplice was found at [[$_BUILDTIME_OSPL_HOME]]," \
+      "which is what will actually be used." \
+      "Manually modify the environment now if this is not the configuration you want." 1>&2
```

Comment:
Apologies since I believe I asked this before but now I cannot find the link. What environment change would be sufficient to force the usage of the runtime OSPL_HOME variable over a valid _BUILDTIME_OSPL_HOME? As far as I can tell nothing other than moving or removing the _BUILDTIME_OSPL_HOME path would be sufficient.

Reply:
They have to modify the environment _afterwards_ e.g. by sourcing a specific `release.com` script manually (and optionally setting the `OSPL_URI` manually). Since users see this message output in the console (usually because they ran `source install_isolated/local_setup.bash`), I put "now" trying to suggest _afterwards_ but maybe "now that this script has been run" or something would be clearer (the challenge is that users don't really know what script is causing this output, it's a bit "magic", so that might confuse them).

Reply:
Okay. I think I have a better understanding now:

This script will override an OSLP_HOME variable set prior to sourcing it if the _BUILDTIME_OSPL_HOME path exists but can be overridden itself by changing that variable afterwards.

Do I have it?

Reply:
yes, that's correct. They probably want to source the `release.com` script too, to setup whatever else (in addition to the `OSPL_HOME` env var). But the simplest option is probably to just re-build the package so that it updates the buildtime path in this script